src/models/bugfindergame.ts

```diff
@@ -86,14 +32,40 @@ export class BugFinderGame {
    *
    * @throws {Error} when player already submitted bug for this code
    *
-   * @returns true when found right bug word and false when not
+   * @returns a list with mistakes in the code
    */
-  public submitWrongCode(bugWordId: number): boolean {
+  public submitWrongCode(submittedSolution: ISolution): Array<boolean> {
     if (this.hasSubmitted()) {
       throw Error('You already submitted this code.');
     }
-    this.solved[this.currentCodeNumber] = this.currentCode.bugWordId == bugWordId;
-    return this.solved[this.currentCodeNumber];
+    const playerBugs: IBug[] = submittedSolution.bugs;
+    const realSolution: ISolution = solutions[this.currentCodeNumber];
+    const bugs: IBug[] = realSolution.bugs;
+
+    const solutionFeedback: Array<boolean> = [];
+
+    bugs.forEach((bug) => {
+      const playerSpecificBug = playerBugs.find((playerBug) => playerBug.wordId == bug.wordId);
+
+      if (playerSpecificBug == null) {
+        solutionFeedback[bug.wordId] = false;
+      } else if (playerSpecificBug.errorType != bug.errorType) {
+        solutionFeedback[bug.wordId] = false;
+      } else {
+        solutionFeedback[bug.wordId] = true;
+      }
```

Comment:
```suggestion
      solutionFeedback[bug.wordId] = playerSpecificBug != null && playerSpecificBug.errorType == bug.errorType;
```

Reply:
In the actual state your code would be the better solution but the error types are till now not fully implemented. In future (Gamify-IT/issues#75) we will enable the selection of the error type and also have to enhance the feedback given. Therefore one boolean isn't enough and we use the if - else-if - else construct to add more feedback.